src/tactic/linarith/frontend.lean

```diff
@@ -149,7 +149,7 @@ Otherwise returns `none`.
 meta def apply_contr_lemma : tactic (option (expr × expr)) :=
 do t ← target,
    match get_contr_lemma_name_and_type t with
-   | some (nm, tp) := do applyc nm, v ← intro1, return $ some (tp, v)
+   | some (nm, tp) := do refine ((expr.const nm []) pexpr.mk_placeholder), v ← intro1, return $ some (tp, v)
```

Comment:
Is this a problem if `nm` depends on universe levels? Or does the elaborator correctly inserts universe levels when none are given like this?

Reply:
No, constants in `pexpr`s don't need universe levels. They'll get added by the elaborator. The test example works over `Type 10`. There's some funny business if you make `α : Type u`, but it's in the elaborator/type class system, before `linarith` is called.